bombastic/api/src/lib.rs

```diff
@@ -31,10 +32,13 @@ pub struct Run {
 }
 
 impl Run {
-    pub async fn run(mut self) -> anyhow::Result<ExitCode> {
-        let state = self.configure()?;
+    pub async fn run(self) -> anyhow::Result<ExitCode> {
+        let index = self.index;
+        let storage = self.storage;
+        let devmode = self.devmode;
```

Comment:
What's the thinking here? Why pass these to `configure` now rather than munging `self`? I ask because this change greatly impacted my #202 PR. :sob:

Reply:
Sorry about that! The reason was that

1. self.configure() would be the place to pass on the &Registry
2. self.infra.run() moves out of self, which makes it impossible to run self.configure() within the async block
3. We get the metrics registry handle in the callback from self.infra.run(), so we have to run configure within the async block, and not reference self.

Reply:
Rebasing #202 isn't trivial, so if you have any insight on how to integrate both approaches, I'm all ears!